### ISSUE TITLE
The putObject endpoint doesn't tag objects with multiple tags correctly

### ISSUE THREAD

**Issue by mattdornfeld** [adobe/S3Mock#673]
I'm using the AWS Java 2 SDK to put an object in s3 with multiple tags. To do this I'm forming a `PutObjectRequest` with the `tagging` field populated. After doing some digging I've found that the AWS API expects multiple tags to be delimited by the & symbol. So the `tagging` field will be a string that looks like
```
key1=value1&key2=value2&key3=value3&...
```
When I attempt to write such a request to s3mock and retrieve the tags on the object I get a single tag that looks like
```
"key1" -> "value1&key2"
```
The first tag is malformed and none of the other ones are returned. I imagine something is going wrong with how s3mock processes the `tagging` field string.

**Comment by afranken** [comment]
Looking into this, we are using the default `Converter<String, List<?>>` provided by Spring which seems to expect some other delimiter like  a comma `,` or a semicolon `;`, but does not split the incoming String by `&`.

Conveniently, all our tests using a `PutObjectRequest` were testing one tag only, so all of then worked... :(

I added another test, and implemented a `Converter` that properly handles the ampersand delimiter, see #713

**Comment by mattdornfeld** [comment]
Appreciate it!

**Comment by afranken** [comment]
Just released `2.4.14` with the fix. :)
